### ISSUE TITLE
linkify is escaping valid HTML tags

### ISSUE THREAD

**Issue by Deimos** [mozilla/bleach#409]
Originally noticed here in relation to the `<abbr>` tag: https://github.com/mozilla/bleach/issues/400

linkify is now using a list of valid HTML tags that is largely built up from constants in html5lib, here: https://github.com/mozilla/bleach/blob/master/bleach/html5lib_shim.py#L49

However, these constants don't seem to make up a complete list of valid tags&mdash;in addition to `<abbr>` (which is now being appended to the list by Bleach), there are also multiple other tags missing, including `del`, `ins`, `sub`, `sup`, and `span`.

It would probably be best to replace this combination of html5lib constants with a full list of valid tags. I can do a PR for this.

**Comment by willkg** [comment]
For anyone who's blocked on this, you can go back to the original pre-3.0 linkify behavior by using `LinkifyFilter` in your own parsing pipeline.

It's something like this:

```
from bleach import html5lib_shim
from bleach.linkify import Linkifyfilter


class OldLinker:
        def __init__(self, callbacks=DEFAULT_CALLBACKS, skip_tags=None, parse_email=False,
            url_re=URL_RE, email_re=EMAIL_RE):
        self.callbacks = callbacks
        self.skip_tags = skip_tags
        self.parse_email = parse_email
        self.url_re = url_re 
        self.email_re = email_re 
        self.parser = html5lib_shim.HTMLParser(namespaceHTMLElements=False) 
        self.walker = html5lib_shim.getTreeWalker('etree') 
        self.serializer = html5lib_shim.HTMLSerializer(
            quote_attr_values='always', 
            omit_optional_tags=False, 
            # linkify does not sanitize
            sanitize=False,
            # linkify alphabetizes
            alphabetical_attributes=False,
        )

    def linkify(self, text):
        dom = self.parser.parseFragment(text)
        filtered = LinkifyFilter(
            source=self.walker(dom),
            callbacks=self.callbacks,
            skip_tags=self.skip_tags,
            parse_email=self.parse_email,
            url_re=self.url_re,
            email_re=self.email_re,
        )
        return self.serializer.render(filtered)
```

**Comment by willkg** [comment]
With #412 merged, I want to additionally tweak the `Linker` to have an additional argument specifying the list of tags it knows about. I'll do that in a followup PR.
